### PR TITLE
Fix arguments for cumulative constraint (again)

### PR DESCRIPTION
**Pull request by smcheema**
Demands are supposed to IntVars actually; make the necessary
testing grammar adjustments as well.

While here, also fix the count of constants within model.String().

### CODE REVIEW

File: internal/testutils/parser/ast/ast.go [hunk 1]
```diff
@@ -133,9 +133,9 @@ func (l *LinearExpr) String() string {
 //   IntervalDemand = Identifier ":" Number .
```
Comment (irfansharif) [inline internal/testutils/parser/ast/ast.go]: Change Number to Identifier.

File: internal/testutils/parser/grammar.ebnf [hunk 2]
```diff
@@ -15,7 +15,7 @@ Variable       = Identifier | Letter "to" Letter .
 Interval       = Identifier "as" "[" Identifier "," Identifier "|" Identifier "]" .
 LinearTerm     = ( [ Digits ] Identifier ) | Digits .
 LinearExpr     = [ "-" ] LinearTerm { ( "+" | "-" ) LinearTerm } | "Σ" "(" Variables ")" .
-IntervalDemand = Identifier ":" Number .
+IntervalDemand = Identifier ":" Variable .
```
Comment (irfansharif) [inline internal/testutils/parser/grammar.ebnf]: Should actually be Identifier instead. Variables are allowed to be of the form "a to b" which we don't want here.

File: internal/testutils/parser/parser.go [hunk 3]
```diff
@@ -202,7 +202,7 @@ func (p *Parser) IntervalDemand() *ast.IntervalDemand {
 	demand := &ast.IntervalDemand{}
 	demand.Name = p.Identifier()
 	p.eat(token.COLON)
-	demand.Demand = p.Number()
+	demand.Demand = p.Variable()
```
Comment (irfansharif) [inline internal/testutils/parser/parser.go]: This will change to Identifier().

File: internal/testutils/parser/testdata/statements [hunk 4]
```diff
@@ -79,6 +79,6 @@ constrain.non-overlapping([i, j], [k, l], true)
 constrain.non-overlapping([i, j], [k, l], true)
 
 statement
-constrain.cumulative(i:3, j:4 | C)
+constrain.cumulative(i:dA, j:dB | C)
```
Comment (irfansharif) [inline internal/testutils/parser/testdata/statements]: Could just use `d` here for both, this just checks that we can parse.

File: internal/testutils/parser/testdata/units [hunk 5]
```diff
@@ -164,14 +164,14 @@ linear-expr
 a + b + c + d
 
 interval-demand
-x: 1
+x: dA
```
Comment (irfansharif) [inline internal/testutils/parser/testdata/units]: Same here, just use `d`.

File: testdata/cumulative [hunk 6]
```diff
@@ -4,8 +4,11 @@ model.vars(sA, sB, sC in [0, 6])
 model.vars(eA, eB, eC in [4, 10])
 model.constants(size == 4)
 model.constants(cap == 6)
+model.constants(dA == 1)
+model.constants(dB == 2)
+model.constants(dC == 3)
 model.intervals(iA as [sA,eA|size], iB as [sB,eB|size], iC as [sC,eC|size])
-constrain.cumulative(iA: 1, iB: 2, iC: 3 | cap)
+constrain.cumulative(iA: dA, iB: dB, iC: dC | cap)
```
Comment (irfansharif) [inline testdata/cumulative:11]: (These are fine to leave as is)

File: testdata/print [hunk 7]
```diff
@@ -3,6 +3,8 @@ model.name(m)
 model.vars(a to c in [0, 10])
 model.literals(x to z)
 model.constants(C == 42)
+model.constants(dA == 1)
```
Comment (irfansharif) [inline testdata/print]: Just use a single `d` here.

File: testdata/recognize [hunk 8]
```diff
@@ -11,7 +11,7 @@ constrain.binary-op(a / b == c)
 constrain.boolean-and(a, b, c to f) if d, e
 constrain.boolean-or(a, b, c to f)
 constrain.boolean-xor(a, b, c to f)
-constrain.cumulative(i: 12, j: 13 | C)
+constrain.cumulative(i: dA, j: dB | C)
```
Comment (irfansharif) [inline testdata/recognize]: Just use single `d` here.

**Comment by musanimu** [comment]
Please change the data_driven.go file with the changed newCumalative Constraint function parameter, You will be passing a string from the data_driven.go function where as its supposed to accept an IntVar now

**Review by irfansharif — CHANGES_REQUESTED** [review]
Mostly looks good to me. Just a few minor suggestions.

**Review by irfansharif — COMMENTED** [review]
LGTM, can you squash all your commits and improve your commit message?

**Comment by smcheema** [comment]
@irfansharif done

**Review by irfansharif — APPROVED** [review]
Thanks!

**Pull request merged**